src/NServiceBus.AcceptanceTests/Core/FakeTransport/ProcessingOptimizations/When_using_concurrency_limit.cs

```diff
@@ -36,14 +37,21 @@ public ThrottledEndpoint()
 
         class FakeReceiver : IPushMessages
         {
+            PushSettings pushSettings;
+
             public Task Init(Func<MessageContext, Task> onMessage, Func<ErrorContext, Task<ErrorHandleResult>> onError, CriticalError criticalError, PushSettings settings)
             {
+                pushSettings = settings;
                 return Task.FromResult(0);
             }
 
             public void Start(PushRuntimeSettings limitations)
             {
-                Assert.AreEqual(10, limitations.MaxConcurrency);
+                if (pushSettings.InputQueue == Conventions.EndpointNamingConvention(typeof(ThrottledEndpoint)))
+                {
+                    // The LimitMessageProcessingConcurrencyTo setting only applies to the input queue
+                    Assert.AreEqual(10, limitations.MaxConcurrency);
```

Comment:
I know we had this assert before. Shouldn't take the chance here to refactor it so that we capture the push settings and then assert in the test the one which belongs to the throttled endpoint instead of inside the fake receiver?